numpy/lib/type_check.py

```diff
@@ -252,6 +252,10 @@ def isreal(x):
     If element has complex type with zero complex part, the return value
     for that element is True.
 
+    Returns True for non-numeric input elements, since the input is
+    interpreted to be an object array for which imaginary units of a complex 
+    number is not defined.
```

Comment:
I am wondering if we should move this into the (currently not existing) `Notes` section.  Or alternatively mark with a `.. note::` markup if we want to make it a proper warning?  Or maybe both: Write in the notes section that a non-complex array returns all `True`, and then add a `.. note::` or `.. warning::` markup because for "object" that is a bit weird (if you have an object array containing complex numbers it is arguably wrong!)

The example is probably good, but should have a short comment before it to explain the context I think.

Reply:
Thanks for the pointers! I've modified the PR accordingly https://github.com/numpy/numpy/pull/18843/files.